Updating to metrics=0.24

gjz010 opened the pull request:
Bumped version of `metrics` library to latest 0.24, as well as making several refactors:

- Splitting library into several files.
- Added some quality-of-life tools (devenv, pre-commit hooks).
- Added `Cargo.lock` to version control.

Bathtor commented on .envrc:
This doesn't need to be checked in.

Bathtor commented on flake.lock:
This doesn't need to be checked in.

Bathtor commented on flake.nix:
This doesn't need to be checked in

Bathtor reviewed (changes requested):
This is really hard to review like this, because it's a lot of changes at once. Can you separate out the refactoring from the version upgrade?

gjz010 commented:
> This is really hard to review like this, because it's a lot of changes at once. Can you separate out the refactoring from the version upgrade?

I will try to reorganize the changes to just keep the necessary changes for upgrading `metrics` version.

gjz010 committed:
dependency: upgrade metrics version to 0.24

gjz010 committed:
fix: add Cargo.lock to version control.

https://github.com/rust-lang/cargo/issues/315
https://blog.rust-lang.org/2023/08/29/committing-lockfiles.html

gjz010 committed:
fix: doc link text

gjz010 commented:
I think the changes by now are the necessary ones.

Bathtor reviewed (approved):
That looks good, thanks.
